test/integration-no-mock/config-management/service-config-push.js

```diff
@@ -36,142 +34,295 @@ module.exports = () => {
-      }
-    ], done);
+    it('modifying secret and bumping local version on internal flex service should succeed', function (done) {
+      this.timeout(250000);
+
+      const initialServiceConfig = {
+        configType: 'service',
+        schemaVersion: '1.0.0',
+        type: 'flex-internal',
+        secret: '123',
+        description: 'Test service',
+        sourcePath: projectPath
+      };
+
+      const modifiedServiceConfig = cloneDeep(initialServiceConfig);
+      modifiedServiceConfig.secret = '456';
+
+      const serviceName = randomStrings.plainString();
+      const initialPkgJson = {
+        version: '1.0.0',
+        dependencies: {
+          'kinvey-flex-sdk': '3.0.0'
+        }
+      };
+      const modifiedPkgJson = cloneDeep(initialPkgJson);
+      modifiedPkgJson.version = '1.0.1';
+
+      async.series([
+        (next) => {
+          ConfigManagementHelper.service.createFromConfig(serviceName, initialServiceConfig, 'org', 'CliOrg', initialPkgJson, (err, id) => {
+            if (err) {
+              return next(err);
+            }
+
+            serviceId = id;
+            next();
+          });
+        },
+        (next) => {
+          // ensure initial flex project is deployed, otherwise the second deploy can fail
+          ConfigManagementHelper.service.assertFlexServiceStatusRetryable(serviceId, initialPkgJson.version, 'ONLINE', next);
+        },
+        (next) => {
+          ConfigManagementHelper.service.modifyFromConfig(serviceId, modifiedServiceConfig, modifiedPkgJson, next);
+        },
+        (next) => {
+          ConfigManagementHelper.service.assertFlexService(serviceId, modifiedServiceConfig, serviceName, next);
+        },
+        (next) => {
+          ConfigManagementHelper.service.assertFlexServiceStatusRetryable(serviceId, modifiedPkgJson.version, null, next);
+        }
+      ], done);
+    });
   });
 
-  it('modifying only description on internal flex service with sourcePath should succeed', (done) => {
-    const initialServiceConfig = {
-      configType: 'service',
-      schemaVersion: '1.0.0',
-      type: 'flex-internal',
-      secret: '123',
-      description: 'Test service',
-      sourcePath: projectPath
-    };
-
-    const modifiedServiceConfig = cloneDeep(initialServiceConfig);
-    modifiedServiceConfig.description = 'Updated service without re-deploying';
-
-    const serviceName = randomStrings.plainString();
-    const pkgJson = {
-      version: '1.0.0',
-      dependencies: {
-        'kinvey-flex-sdk': '3.0.0'
-      }
-    };
-
-    async.series([
-      (next) => {
-        ConfigManagementHelper.service.createFromConfig(serviceName, initialServiceConfig, 'org', 'CliOrg', pkgJson, (err, id) => {
-          if (err) {
-            return next(err);
+  describe('rapid data', () => {
+    it('adding mapping to a mssql service should succeed', (done) => {
+      const serviceName = randomStrings.plainString();
+      const initialServiceConfig = {
```

Comment:
I see a lot of service configurations in the tests. Isn't it better to have them outside the code in files?

Reply:
Me and @tsvetomir-nedyalkov, we've discussed it and decided that it will be better if they are in the code. Things seem more flexible like this and I believe that at some point we could refactor the code and decrease the service configurations (by using some functions that will build what is needed). Also, if we use files, they could grow a lot when we add more tests and at the same time some of them could differ just by a property or two.

Reply:
After an offline discussion, me, Denny and Tsvetomir decided that we will leave it like that for now. We will later create build functions for config files in the tests that will reduce the inline JSON.